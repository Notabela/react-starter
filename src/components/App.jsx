import React from 'react';
import { ReactComponent as Logo } from 'assets/logo.svg';
import { add } from 'utils';
import 'styles/App.scss';

const App = () => {
  const [value, setValue] = React.useState("React");

  return (
    <div className="App">
    <header className="App-header">
      <Logo className="App-logo" />
      <h1 className="App-win">{ add('This is', value) }!</h1>
      <button
        className="App-button"
        onClick={ () => setValue({ React: 'CSC 59939', 'CSC 59939': 'React' }[value]) }
      >
        Toggle
      </button>
    </header>
    </div>
  )
}

export default App;
