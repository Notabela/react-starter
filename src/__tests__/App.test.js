import { render, fireEvent } from '@testing-library/react';
import App from 'components/App';

// two-kinds of testing - snapshot testing, and DOM testing

describe('Test App', () => {
  // snapshot testing
  it('matches home pages snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  })

  // DOM testing -> render components and test them
  // testing library is different from common libraries
  //  The utilities this library provides facilitate querying the DOM in the same way the user would. Finding form elements by their label text (just like a user would),
  // finding links and buttons from their text (like a user would).
  it('toggles text correctly', () => {
    const { getByRole, getByText } = render(<App />);
    expect(getByRole('heading', { level: 1})).toHaveTextContent(/This is React/i) // i = case-insensitive

    fireEvent.click(getByRole('button'))

    const headerElement = getByText(/This is CSC 59939/i)
    expect(headerElement).toBeInTheDocument();
  })
})
