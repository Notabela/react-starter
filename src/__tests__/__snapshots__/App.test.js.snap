// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App matches home pages snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <svg
        class="App-logo"
      >
        logo.svg
      </svg>
      <h1
        class="App-win"
      >
        This is React!
      </h1>
      <button
        class="App-button"
      >
        Toggle
      </button>
    </header>
  </div>
</DocumentFragment>
`;
